src/gentropy/dataset/test.py

```diff
@@ -0,0 +1,55 @@
+"""Study locus dataset."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from pyspark.sql.types import (
+    DoubleType,
+    LongType,
+    StringType,
+    StructField,
+    StructType,
+)
+
+from gentropy.dataset.dataset import Dataset
+
+
+@dataclass
+class TestClass(Dataset):
+    """Plain test class demonstrating the use of automatic  ID generation and instantiation without schema."""
+
+    # This list defines the uniqueness of the dataset - these have to be mandatory fields:
+    _unique_fields = ["a", "b", "c"]
+    _id_column = "testId"
+
+    def __post_init__(self: Dataset) -> None:
```

Comment:
Why doing it in the child class?

This could be set in the parent class __post_init__

In theory you overwrite the parameters to the `generate_identifier` that is called in the Dataset.__post_init__.
The method should infer three things:
1. weather the dataset requires the index field
2. index field name
3. fields that are used to build the index field.

These 3 things should be defined in child classes, in Dataset class they should be dummy field placeholders.

The only thing to implement then is to run the `super().__post_init__` in case we overwrite the __post_init_ directly in the child class.